lib/thread.c

```diff
@@ -974,6 +974,8 @@ static void thread_cancel_rw(struct thread_master *master, int fd, short state)
 		memmove(master->handler.pfds + i, master->handler.pfds + i + 1,
 			(master->handler.pfdcount - i - 1)
 				* sizeof(struct pollfd));
+		master->handler.pfds[master->handler.pfdcount].fd = 0;
+		master->handler.pfds[master->handler.pfdcount].events = 0;
```

Comment:
Don't these lines have to go after the count decrement?

Reply:
You are right.  In my first fix I was doing it when I was re-using the index but in second one after review comment I submitted, I used diff provided but here the count has to be decremented before resetting for last index as array index start with 0 and the index that needs to be cleaned up is the one after decremented.

Reply:
I have tested my script again with latest diff and it works fine.